# prediction horizon
dt: 0.02
horizon: 50

# parameters for the crazyflie model
quadmodel:
  mass: 0.028
  arm_length: 0.044
  Ixx: 2.3951e-5
  Iyy: 2.3951e-5
  Izz: 3.2347e-5
  cm: 2.4e-6
  tau: 0.08

# weights for the cost function
cost_weights:
  x: 20.0
  y: 20.0
  z: 20.0
  vx: 2.0
  vy: 2.0
  vz: 2.0
  roll: 1.0
  pitch: 1.0
  yaw: 1.0
  roll_c: 1.0
  pitch_c: 1.0
  yaw_c: 1.0
  thrust: 1.0

# input constraints
input_constraints:
  roll_min: -0.25
  roll_max: 0.25
  pitch_min: -0.25
  pitch_max: 0.25
  yaw_min: -0.1
  yaw_max: 0.1
  thrust_min: 0.0
  thrust_max: 0.5

# state constraints
state_constraints:
  x_min: -10
  x_max: 10
  y_min: -10
  y_max: 10
  z_min: 0
  z_max: 4.0
  vx_min: -2
  vx_max: 2
  vy_min: -2
  vy_max: 2
  vz_min: -2
  vz_max: 2
  roll_min: -0.25
  roll_max: 0.25
  pitch_min: -0.25
  pitch_max: 0.25
  yaw_min: -3.14
  yaw_max: 3.14